pkg/validations/cluster.go

```diff
@@ -79,3 +79,26 @@ func ValidateK8s126Support(clusterSpec *cluster.Spec) error {
 	}
 	return nil
 }
+
+// ValidateManagementClusterBundlesVersion checks if management cluster's bundle version is greater than or equal to the bundle version used to upgrade a workload cluster.
+func ValidateManagementClusterBundlesVersion(ctx context.Context, k KubectlClient, mgmtCluster *types.Cluster, clusterSpec *cluster.Spec) error {
+	cluster, err := k.GetEksaCluster(ctx, mgmtCluster, mgmtCluster.Name)
+	if err != nil {
+		return err
+	}
+
+	if cluster.Spec.BundlesRef == nil {
+		return fmt.Errorf("management cluster bundlesref cannot be nil")
+	}
+
+	mgmtBundles, err := k.GetBundles(ctx, mgmtCluster.KubeconfigFile, cluster.Spec.BundlesRef.Name, cluster.Spec.BundlesRef.Namespace)
+	if err != nil {
+		return err
+	}
+
+	if mgmtBundles.Spec.Number < clusterSpec.Bundles.Spec.Number {
```

Comment:
if not specifying bundleOverride flag, will the workload `clusterSpec` here has a valid bundle number? like does the clusterSpec being properly built with bundle workload cluster bundle fetched from existing mgmt cluster?